.github/workflows/ruby.yml

```diff
@@ -0,0 +1,43 @@
+name: Ruby CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        exclude:
+          - ruby-version: 2.3.8
+            gemfile: gemfiles/activesupport_6.gemfile
+          - ruby-version: 2.4.10
+            gemfile: gemfiles/activesupport_6.gemfile
+        gemfile:
+          - gemfiles/activesupport_5.gemfile
+          - gemfiles/activesupport_6.gemfile
+        ruby-version:
+          - 2.3.8
+          - 2.4.10
+          - 2.5.8
+          - 2.6.6
+          - 2.7.2
+          - '3.0.0' # Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'
```

Comment:
If a clean run works, we should probably change this to `'3.0' to test any patch versions automatically.